pkg/ring/model.go

```diff
@@ -127,6 +133,16 @@ func (d *Desc) TokensFor(id string) (tokens, other Tokens) {
 	return myTokens, takenTokens
 }
 
+// GetRegisteredAt returns the timestamp when the instance has been registered to the ring
+// or a zero value if unknown.
+func (i *IngesterDesc) GetRegisteredAt() time.Time {
+	if i == nil {
```

Comment:
```suggestion
	if i == nil || i.RegisteredTimestamp == 0 {
```

(otherwise time.Unix(0, 0) will not be zero time.)

Reply:
Good point. Fixed and added a test.